chart/etcd-backup-restore/templates/etcd-statefulset.yaml

```diff
@@ -0,0 +1,241 @@
+        - --storage-provider={{ .Values.backup.storageProvider }}
+        - --store-prefix={{ .Release.Name }}-etcd
+{{- if .Values.backup.etcdQuotaBytes }}
+        - --embedded-etcd-quota-bytes={{ int $.Values.backup.etcdQuotaBytes }}
+{{- end }}
+{{- if .Values.tls }}
+        - --cert=/var/etcd/ssl/tls/tls.crt
+        - --key=/var/etcd/ssl/tls/tls.key
+        - --cacert=/var/etcd/ssl/ca/ca.crt
+        - --insecure-transport=false
+        - --insecure-skip-tls-verify=false
+        - --endpoints=https://{{ .Release.Name }}-etcd-0:2379
+{{ else }}
+        - --insecure-transport=true
+        - --insecure-skip-tls-verify=true
+        - --endpoints=http://{{ .Release.Name }}-etcd-0:2379
+{{- end }}
+        - --etcd-connection-timeout={{ .Values.backup.etcdConnectionTimeout }}
+        - --delta-snapshot-period-seconds={{ int $.Values.backup.deltaSnapshotPeriodSeconds }}
+        - --delta-snapshot-memory-limit={{ int $.Values.backup.deltaSnapshotMemoryLimit }}
+        image: {{ .Values.images.etcdBackupRestore.repository }}:{{ .Values.images.etcdBackupRestore.tag }}
+        imagePullPolicy: {{ .Values.images.etcdBackupRestore.pullPolicy }}
+        ports:
+        - containerPort: 8080
+          name: server
+          protocol: TCP
+        resources:
+{{ toYaml .Values.resources.backup | indent 10 }}
+        env:
+        - name: STORAGE_CONTAINER
+          value: {{ .Values.backup.storageContainer }}
+{{- if eq .Values.backup.storageProvider "S3" }}
+        - name: "AWS_REGION"
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Release.Name }}-etcd-backup
+              key: "region"
+        - name: "AWS_SECRET_ACCESS_KEY"
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Release.Name }}-etcd-backup
+              key: "secretAccessKey"
+        - name: "AWS_ACCESS_KEY_ID"
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Release.Name }}-etcd-backup
+              key: "accessKeyID"
+{{- else if eq .Values.backup.storageProvider "ABS" }}
+        - name: "STORAGE_ACCOUNT"
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Release.Name }}-etcd-backup
+              key: "storageAccount"
+        - name: "STORAGE_KEY"
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Release.Name }}-etcd-backup
+              key: "storageKey"
+{{- else if eq .Values.backup.storageProvider "GCS" }}
+        - name: "GOOGLE_APPLICATION_CREDENTIALS"
+          value: "/root/.gcp/serviceaccount.json"
+{{- else if eq .Values.backup.storageProvider "Swift" }}
+        - name: "OS_AUTH_URL"
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Release.Name }}-etcd-backup
+              key: "authURL"
+        - name: "OS_DOMAIN_NAME"
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Release.Name }}-etcd-backup
+              key: "domainName"
+        -	name: "OS_USERNAME"
```

Comment:
@swapnilgm thanks for making the changes. Just one more small correction - there is a tabspace here. Please change it to a single space.

Reply:
done